### ISSUE TITLE
Tab is destroyed when detached if it has previously been discarded

### ISSUE THREAD

**Issue by petemill** [brave/muon#496]
Even if the tab has been reloaded since it was discarded, detach will result in a destroyed tab.

STR:

1. Create a tab, load contents, and make a different tab active
1. `tab.discard()` on the original tab
  - the tab ID will be changed
2. Make the tab active so that it reloads it's content
   - the tab ID will stay the same
3. `tab.detach()` on the new web contents
4. The tab will be destroyed

Event log:
```
tab[33].detach(): { windowId: 1, guestInstance: 6 }
 - [37] on-web-contents-created
 - [33] will-detach
 - [33] tab-replaced-at { newTabId 37, newTab: { windowId: -1, guestInstance: 7 } }
 - [33] will-destroy
 - [33] destroyed
 - [33] render-view-deleted
 - [37] will-attach { windowId: 1 }
 - [33] <webview>.tab-id-changed { tabId: 37 }
 - [37] will-destroy
 - [37] destroyed
new BrowserWindow()-->entry.js-->addFrame({tabId: 33})
// tab not found!
```

Expected event log (real case of a tab that has *never* been discarded:
```
tab[13].detach() current state - { windowId: 1, guestInstance: 2 }
 - [37] web-contents-created
 - [13] will-detach
 - [13] tab-replaced-at { newTabId: 37, newTab: { windowId: -1, guestInstance: 7 }
 - [13] <webview>.tab-id-changed { tabId: 37 }
 - [13] did-detach
 - [13] chrome-tabs-updated { index: -1 }
 - [37] will-attach { windowId: 1 }
 - [37] will-destroy
 - [37] destroyed
 - [37] render-view-deleted (x2)
 - [13] <webview>.tab-id-changed { tabId: 37 }
new BrowserWindow()-->entry.js-->addFrame({tabId: 13})
 - [13] will-attach
 - [13] tab-inserted-at
 - [13] set-active
 - [13] did-attach { windowId: 4}
 - [13] render-view-ready
```
